### PR TITLE
QUnit: Rework API for ignoring uncleared timers

### PR DESCRIPTION
**Pull request by San4es**
See [commits](https://github.com/DevExpress/DevExtreme/pull/11765/commits) for more details.

### CODE REVIEW

**Review by DokaRus — APPROVED** [review]

File: testing/helpers/qunitExtensions.js [hunk 1]
```diff
@@ -137,13 +137,30 @@
         });
     };
 
+    const beforeTestDoneCallbacks = [];
+
+    QUnit.beforeTestDone = function(callback) {
```
Comment (nightskylark) [inline testing/helpers/qunitExtensions.js:142]: Why you are not using arrow functions here?

File: testing/helpers/qunitExtensions.js [hunk 2]
```diff
@@ -363,176 +366,121 @@
...
-                    skip = true;
-                    return false;
-                }
-            });
+    const logTestFailure = function(timerInfo) {
+        const timeoutString = timerInfo.timeout ? '\nTimeout: ' + timerInfo.timeout : '';
 
-            return skip;
-        };
+        const message = [
```
Comment (nightskylark) [inline testing/helpers/qunitExtensions.js:435]: Why don't use a multiline string?
```
        const message = `Not cleared timer detected.
Timer type: ${timerInfo.timerType}
Id: ${timerInfo.timerId}
Callback:
${timerInfo.callback}
${timeoutString}`;
```

Comment (San4es) [reply testing/helpers/qunitExtensions.js:435]: Interpolated strings are not supported in IE:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: testing/helpers/qunitExtensions.js (hunk 1 above)
Comment (San4es) [reply testing/helpers/qunitExtensions.js:142]: Arrow functions are not supported in IE:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

**Pull request merged**